demopacket.go

```diff
@@ -15,14 +14,60 @@ import (
 // See also: ParserConfig.AdditionalNetMessageCreators & Parser.RegisterNetMessageHandler()
 type NetMessageCreator func() proto.Message
 
+func NewNetMessageFunc(t interface{}) (int, func() proto.Message) {
```

Comment:
I've used it for making it easier to create NET message handlers, e.g.

```go
cfg := dem.DefaultParserConfig

cvarID, cvarFunc := dem.NewNetMessageFunc(msg.CNETMsg_SetConVar{})
cfg.AdditionalNetMessageCreators = map[int]dem.NetMessageCreator{
	cvarID: cvarFunc,
}

p := dem.NewParserWithConfig(rd, cfg)
```

Instead of

```go
cfg := dem.DefaultParserConfig
cfg.AdditionalNetMessageCreators = map[int]dem.NetMessageCreator{
	4: func() proto.Message { return new(msg.CNETMsg_Tick) },
}

p := dem.NewParserWithConfig(f, cfg)

```
It's not a pretty implementation, and it's not something that I actually meant to include in the PR.

But I do believe that it should be a bit easier to add NET handlers :slightly_smiling_face: But that should probably be done in a different thread!

Reply:
Ok, yeah we can look into this separately.

I'm not 100% sure it's necessary - but anyway, topic for another PR / Issue.